packages/account/src/predicate/predicate.ts

```diff
@@ -80,6 +80,26 @@ export class Predicate<
     }
   }
 
+  /**
+   * Creates a new Predicate instance with updated parameters while maintaining the original bytecode and ABI
+   *
+   * @param params - Object containing optional new configurableConstants and data
+   * @returns A new Predicate instance with the updated parameters
+   */
+  toNewInstance(params: {
+    configurableConstants?: TConfigurables;
+    data?: TData;
+  }): Predicate<TData, TConfigurables> {
+    return new Predicate<TData, TConfigurables>({
+      bytecode: this.bytes,
+      abi: this.interface?.jsonAbi,
+      provider: this.provider,
+      data: params.data ?? this.predicateData,
+      configurableConstants: params.configurableConstants,
+      loaderBytecode: this.loaderBytecode,
```

Comment:
```suggestion
```

The `loaderBytecode` is no longer a field on the `Predicate` class (after https://github.com/FuelLabs/fuels-ts/pull/3389).